src/components/AuthContext.js

```diff
@@ -0,0 +1,100 @@
+import React, { useEffect, useState } from 'react'
+import { withRouter } from 'react-router-dom'
+import { useHistory } from 'react-router-dom'
+import { auth, db } from '../firebase'
+
+const AuthContext = React.createContext()
+
+function AuthProvider({ children, setProfile, setTasks, setPets }) {
+  const [user, setUser] = useState({})
+  const history = useHistory()
+
+  useEffect(() => {
+    auth.onAuthStateChanged((user) => {
+      if (user) {
+        setUser({
+          id: user.uid,
+          email: user.email,
+        })
+        window.localStorage.setItem('uid', user.uid)
+        getUserInformation()
+        const currentUser = auth.currentUser.uid
+        getPets(currentUser)
+        getTasks(currentUser)
+      } else {
+        setUser({})
+        setProfile({ email: '', password: '', id: '' })
+        window.localStorage.removeItem('uid')
+      }
+    })
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  async function getUserInformation() {
+    await db
+      .collection('users')
+      .doc(auth.currentUser.uid)
+      .get()
+      .then((doc) => {
+        return doc.exists && doc.data()
+      })
+      .then(async (data) => {
+        await setProfile({ ...data })
+      })
```

Comment:
I might be wrong but couldn't you leave out the async and await here as you are already using .then which is a promise? so just write .then((data) => setProfile({...data}) ?

Reply:
You could be right, I will give that a try.